### PR TITLE
Let boxplot() defer rcParams application to bxp()

### PR DESCRIPTION
**Pull request by anntzer**
instead of duplicating the defaults-applying code in both methods.

This caught an oversight in the application of the
boxplot.boxprops.linewidth rcParam in bxp (see changelog note);
because of that, the bxp testing code needed some update (to keep the
old behavior).

Preliminary work to having bxp() normalize properties passed in the
boxprops, meanprops, etc. kwargs as well (#13448).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -3895,88 +3885,66 @@ def bxp(self, bxpstats, positions=None, widths=None, vert=True,
...
-            linewidth=rcParams['boxplot.flierprops.linewidth'],
-            color=rcParams['boxplot.flierprops.color'],
-            marker=rcParams['boxplot.flierprops.marker'],
-            markerfacecolor=rcParams['boxplot.flierprops.markerfacecolor'],
-            markeredgecolor=rcParams['boxplot.flierprops.markeredgecolor'],
-            markeredgewidth=rcParams['boxplot.flierprops.markeredgewidth'],
-            markersize=rcParams['boxplot.flierprops.markersize'],
-        )
-
+        # cap properties
```
Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: For me, it takes enough mental bandwidth to read these dict-comprehensions that I wonder if factoring out into a sub function makes sense

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -2108,68 +2108,59 @@ def layers(n, m):
     axs[1, 1].stackplot(range(100), d.T, baseline='weighted_wiggle')
 
 
+def _bxp_test_helper(
+        stats_kwargs={}, transform_stats=lambda s: s, bxp_kwargs={}):
```
Comment (phobson) [inline lib/matplotlib/tests/test_axes.py]: this isn't a big deal. feel free to ignore it. but my early morning, uncaffeinated brain has a hard time with function definitions whose first parameters is on a second line

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: pushed a second commit that does it, indeed that's a good idea.

Comment (phobson) [reply lib/matplotlib/axes/_axes.py]: ahhh it looks great.

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -2108,68 +2108,59 @@ def layers(n, m):
...
+    logstats = matplotlib.cbook.boxplot_stats(
+        np.random.lognormal(mean=1.25, sigma=1., size=(37, 4)), **stats_kwargs)
+    fig, ax = plt.subplots()
+    if bxp_kwargs.get('vert', True):
+        ax.set_yscale('log')
+    else:
+        ax.set_xscale('log')
+    # Work around baseline images generate back when bxp did not respect the
+    # boxplot.boxprops.linewidth rcParam when patch_artist is False.
+    if not bxp_kwargs.get('patch_artist', False):
```
Comment (jklymak) [inline lib/matplotlib/tests/test_axes.py]: OK, but you never test the new proper behaviour this way, do you?

Comment (anntzer) [reply lib/matplotlib/tests/test_axes.py]: indeed.  slightly changed the test setup to test that the rc is used.

**Pull request merged**